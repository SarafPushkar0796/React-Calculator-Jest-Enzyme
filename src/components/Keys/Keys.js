import React from 'react';
import PropTypes from 'prop-types';
import './Keys.css';

/**
 * 
 *  The Keys component is also stateless and it receives the following props:
     -keyAction - the function related to the key type.
     -keyType - a string used to determine which CSS rules the Key will have.
     -keyValue - a string used to determine the value to be passed to the keyAction function.
 */
const Keys = ({ 
    keyAction, 
    keyType, 
    keyValue 
}) => (
    <div className={ `key-container ${keyType}` } onClick={() => keyAction(keyValue)}>
        <p className="key-value">
            {keyValue}
        </p>
    </div>
);

Keys.propTypes = {
  keyAction: PropTypes.func.isRequired,
  keyType: PropTypes.string.isRequired,
  keyValue: PropTypes.string.isRequired,
}

export default Keys;