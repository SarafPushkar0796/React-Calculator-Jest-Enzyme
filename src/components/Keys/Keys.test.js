import React from 'react';
import { shallow } from 'enzyme';
import Keys from './Keys';

describe('Keys', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(
            <Keys
                keyAction={jest.fn()}
                keyType={''}
                keyValue={''}
            />
        );
    });

    it('should render a <div />', () => {
        expect(wrapper.find('div').length).toEqual(1);
    });

    it('should render the value of keyValue', () => {
        wrapper.setProps({ keyValue: 'test' });
        expect(wrapper.text()).toEqual('test');
    });

    it('should render correctly', () => expect(wrapper).toMatchSnapshot());
});