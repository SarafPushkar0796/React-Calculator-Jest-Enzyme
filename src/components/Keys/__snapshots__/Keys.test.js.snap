// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Keys should render correctly 1`] = `ShallowWrapper {}`;
